app/scripts/views/mixins/signin-mixin.js

```diff
@@ -55,19 +56,35 @@ define(function (require, exports, module) {
     },
 
     onSignInSuccess: function (account) {
-      if (! account.get('verified')) {
+      if (account.get('verified') && account.get('challenge') === Constants.SIGNIN_EMAIL) {
```

Comment:
It should be `! account.get('verified')` here, that's how we are bootstrapping the challenge into old clients that don't support showing a slick UI. Old clients that only know how to handle the `verified` status will see false and show the user the standard `go confirm your email` screen.